Make network request timeouts configurable (#110)
iqm-finland/iqm-client: INTEGRATION_GUIDE.rst

In some special cases, it makes sense to disable this check by changing the default value of parameter ``circuit_duration_check`` of :meth:`IQMClient.submit_circuits` to ``False``.
Disabling the circuit duration check may be limited to certain users or groups, depending on the server settings. In normal use, the circuit duration check should always remain enabled.

Note on environment variables
-----------------------------

> Set ``IQM_CLIENT_REQUESTS_TIMEOUT`` environment variable to override the network requests default timeout value. Currently it is set to 60 seconds. This might be needed when fetching run results of larger circuits via slow network connections.

Reviewer: ```suggestion
Set ``IQM_CLIENT_REQUESTS_TIMEOUT`` environment variable to override the network requests default timeout value. The default value is 60 seconds and might not be sufficient when fetching run results of larger circuits via slow network connections.
```
